src/recipes/heartbeat-by-user-first-impression/index.js

```diff
@@ -241,10 +241,24 @@ let run = function (state, extras) {
 
   let eUrls = [
     `https://qsurvey.mozilla.com/s3/Firefox-USE-Survey?source=heartbeat&surveyversion=${VERSION}&updateChannel=${state.updateChannel}&fxVersion=${state.fxVersion}`,
-    `http://qsurvey.mozilla.com/s3/PBM-Survey-Genpop-41?source=heartbeat&surveyversion=${VERSION}&updateChannel=${state.updateChannel}&fxVersion=${state.fxVersion}`
+    `https://qsurvey.mozilla.com/s3/PBM-Survey-Genpop-41?source=heartbeat&surveyversion=${VERSION}&updateChannel=${state.updateChannel}&fxVersion=${state.fxVersion}`,
+    `https://qsurvey.mozilla.com/s3/Heartbeat-Bright-Spots?source=heartbeat&surveyversion=${VERSION}&updateChannel=${state.updateChannel}&fxVersion=${state.fxVersion}`
   ];
 
-  let engagementUrl = eUrls[~~(Math.random() <= 0.50)];
+  let cutBreaks = function (arr, breaks, rng=Math.random()) {
```

Comment:
handling all the various error conditions is part of why I think Python and JS don't have a function like this as a 'stock' thing.  

R's cut (http://www.inside-r.org/r-doc/base/cut) is the only good prior art, and it has a ton of flexiblity.  

I waffled on the {weight, url}  vs   Urls, Breaks.  Both are tedious :)  But I htink when we solve surveys *for real*, it will be a list of objects.